jupyterlab_gitsync/src/components/toolbar_btn_template.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+import { classes, style } from 'typestyle';
+import {
+  toolbarItemClass,
+  toolbarButtonClass,
+  toolbarButtonIconClass,
+} from '../style/toolbar';
+
+import { Props } from './panel';
+
+export interface State {
+  label: string;
+  style: ReturnType<typeof style>;
+  options?: any;
```

Comment:
It looks like you're using this `options` field to keep track of state (running vs not running). Wouldn't it be cleaner to use an enum or boolean instead.

Reply:
Since right now the buttons are pretty simple in functionality it may be simpler to do a boolean right now, but since I plan on having the status indicator to be on the toolbar and will most likely be adding more complex buttons that require more state parameters, I decided it might be easier to do just an optional `options` parameter instead of having many optional parameters specific to each individual button. 

For example, the `ControlButton` (currently used to toggle the auto sync feature) needs to keep track of the running state for the button whereas the status indicator will not need to do this, but instead might have a optional `state` parameter indicating which state of the sync process it is in. Since I'm not quite sure if there will need to be more state parameters for specific button, I thought it might be simpler to just use a single `options` parameter encapsulating all of this. 

Let me know what your thoughts are about doing a general `options` parameter vs. having all the optional parameters explicitly stated.

Reply:
I see. I'm actually not too sure what is preferred in javascript/typescript, but generally, it's preferred to have consistent  typing. I that would make your code more robust, but for now, I'm fine with you leaving it as is.

Please do add the description of the intended use from your last post as a code comment in this file for clarification.

Reply:
After talking to Jason, I realize that we can actually get rid of this field once I do the refactoring for the ToolbarButton component. I'll add the description for it for now, but this will be fixed in a future PR.

Reply:
SGTM!